src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1848,3 +1848,9 @@ def validate_policy(namespace):
     if namespace.id is not None:
         logger.warning("\nPlease do not specify --expiry and --permissions if they are already specified in your "
                        "policy.")
+
+
+def validate_blob_name_for_upload(namespace):
```

Comment:
What if invalid blob name is provided? Such as `D:\myfolder` or `D:\myfolder\`  which are actually directories?

Reply:
The original code throws an internal error, should probably throw a warning to the user and do nothing.

Reply:
Have discussed with Yishi to keep behaviors in main repo and extension consistent: throw the necessary exceptions (directory, file permission, etc.)